File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for (String str : strs) {
            char[] key = str.toCharArray();
            Arrays.sort(key);

            // 현재 키에 해당하는 리스트가 없다면 새로 생성, 있다면 기존 리스트에 추가
            map.computeIfAbsent(String.valueOf(key), k -> new ArrayList<>()).add(str);
        }

        // HashMap의 값들을 리스트로 반환
        return new ArrayList<>(map.values());
    }
}

File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map: MutableMap<String, MutableList<String>> = mutableMapOf()

        for (str in strs) {
            val key = str.toCharArray().sorted().joinToString("")
            map.getOrPut(key) { mutableListOf() }.add(str)
        }

        return ArrayList<List<String>>(map.values)
    }
}